projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -0,0 +1,239 @@
+  public onMetricChange(newAttribute: AttributeMetadata): void {
+    this.buildSpecAndEmit(
+      combineLatest([of(newAttribute), this.selectedAggregation$, of(this.orderByExpression?.direction)])
+    );
+  }
+
+  public onAggregationChange(newAggregation: MetricAggregationType): void {
+    this.buildSpecAndEmit(
+      combineLatest([this.selectedMetric$, of(newAggregation), of(this.orderByExpression?.direction)])
+    );
+  }
+
+  public onSortByChange(newSortBy?: GraphQlSortDirection): void {
+    this.buildSpecAndEmit(combineLatest([this.selectedMetric$, this.selectedAggregation$, of(newSortBy)]));
+  }
+
+  private buildSpecAndEmit(
+    orderByData$: Observable<
+      [AttributeMetadata | undefined, MetricAggregationType | undefined, GraphQlSortDirection | undefined]
+    >
+  ): void {
+    orderByData$
+      .pipe(
+        take(1),
+        switchMap(orderByData => this.buildOrderExpression(...orderByData))
+      )
+      .subscribe(value => this.orderByExpressionChange.emit(value));
+  }
+
+  private getCurrentSelectedAttribute(): Observable<AttributeMetadata | undefined> {
+    return this.metricOptions$.pipe(
+      map(attributeOptions =>
+        attributeOptions.find(option => option.value.name === this.orderByExpression?.keyExpression?.key)
+      ),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getCurrentSelectedAggregation(): Observable<MetricAggregationType | undefined> {
+    return this.aggregationOptions$.pipe(
+      map(aggregationOptions =>
+        aggregationOptions.find(option => option.value === this.orderByExpression?.aggregation)
+      ),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getCurrentSelectedSortBy(): Observable<GraphQlSortDirection | undefined> {
+    return this.sortByOptions$.pipe(
+      map(sortByOptions => sortByOptions.find(option => option.value === this.orderByExpression?.direction)),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getAggregationOptionsForAttribute(selected?: AttributeMetadata): SelectOption<MetricAggregationType>[] {
+    if (selected === undefined) {
+      return [];
+    }
+
+    return selected.allowedAggregations.map(aggType => ({
+      value: aggType,
+      label: getAggregationDisplayName(aggType)
+    }));
+  }
+
+  private getAttributeOptionsForContext(context?: TraceType): Observable<SelectOption<AttributeMetadata>[]> {
+    if (context === undefined) {
+      return of([]);
+    }
+
+    return this.metadataService.getSelectionAttributes(context).pipe(
+      map(attributes =>
+        attributes.map(attribute => ({
+          value: attribute,
+          label: this.metadataService.getAttributeDisplayName(attribute)
+        }))
+      )
+    );
+  }
+
+  private getSortByOptions(): Observable<SelectOption<GraphQlSortDirection | undefined>[]> {
+    return of([
+      {
+        value: undefined,
```

Comment:
Should not be an option. But this makes me wonder, what's the process like for removing a sort? maybe some kind of reset icon is needed

Reply:
We're using the 'None' option in other filters like order by. So I just keeping the same logic here.

Reply:
just to clarify, filters don't have that behavior - they have a clear icon. I believe group by does though. If we want to follow that pattern, we should be putting that none option in the attribute selection rather than the direction selector.

Reply:
done